### PR TITLE
fix(schema): Don't default to tracking all nodes

### PR DESCRIPTION
**Pull request by freiksenet**
## Description

We accidentaly made `getAllNodes` track everything when called. It's not an expected behavioru, because this function is often used to manually filter nodes. Thus, we disable tracking unless you explicitly ask for it. This PR also adds manual tracking function.

### CODE REVIEW

File: packages/gatsby/src/schema/node-model.js [hunk 1]
```diff
@@ -142,11 +140,11 @@ class LocalNodeModel {
...
     }
 
-    return trackPageDependencies(
-      result,
-      { path: this.path, ...pageDependencies },
-      this.createPageDependency
-    )
+    if (pageDependencies) {
+      return this.trackPageDependencies(result, pageDependencies)
+    } else {
```
Comment (wardpeet) [inline packages/gatsby/src/schema/node-model.js]: no need to do an else statement

**Comment by stefanprobst** [comment]
LGTM!

File: packages/gatsby/src/schema/node-model.js [hunk 2]
```diff
@@ -224,6 +218,35 @@ class LocalNodeModel {
...
+    const fullDependencies = {
+      path: this.path,
+      ...(pageDependencies || {}),
+    }
+    const { path, connectionType } = fullDependencies
+    if (path) {
+      if (connectionType) {
+        this.createPageDependency({ path, connection: connectionType })
+      } else {
+        const nodes = Array.isArray(result) ? result : [result]
```
Comment (wardpeet) [inline packages/gatsby/src/schema/node-model.js]: This should work too, don't know which one is more readable. (feel free to ignore)
```suggestion
        const nodes = [].concat(result);
```

**Comment by wardpeet** [comment]
This looks good to me. Just to clarify `getAllNodes` is actually the function that fixes this issue right? Could you also explain a bit what this fixes? Can't seem to see an issue attached or a proper explanation why this is good 😛 Listening to fewer nodes is obviously a plus but want to know if there is more to it.

**Pull request merged**